{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "c077e4b4-0589-446d-84db-1b85e37bc1f0",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import shutil\n",
    "\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "\n",
    "from os import listdir\n",
    "from os.path import isfile, join\n",
    "import json \n",
    "\n",
    "import plotly\n",
    "import plotly.express as px\n",
    "import plotly.io as pio"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a0b05a14-44a2-432e-953a-1076176e32d7",
   "metadata": {},
   "source": [
    "# featureCounts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "563e2b64-2d8b-4390-b435-c3e456ad55c0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "metadata_subdir exists:True\n",
      "\n",
      "PEAKS_dir exists:True\n",
      "\n",
      "BAM_DIR exists:True\n",
      "\n"
     ]
    }
   ],
   "source": [
    "experiment_id = \"NS-23.0061\"\n",
    "\n",
    "metadata_subdir = os.path.expanduser(\"~/fht.samba.data/experiments/ATACseq/{}/analysis/metadata/\".format(experiment_id))\n",
    "print('metadata_subdir exists:{}\\n'.format(os.path.exists(metadata_subdir)))\n",
    "\n",
    "PEAKS_dir = os.path.expanduser(\"~/fht.samba.data/experiments/ATACseq/{}/alignment/macs2/\".format(experiment_id))\n",
    "print('PEAKS_dir exists:{}\\n'.format(os.path.exists(PEAKS_dir)))\n",
    "\n",
    "BAM_DIR = os.path.expanduser(\"~/fht.samba.data/experiments/ATACseq/{}/alignment/bam/\".format(experiment_id))\n",
    "print('BAM_DIR exists:{}\\n'.format(os.path.exists(BAM_DIR)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "348eccae-6d03-453f-ba06-458a598861b6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "../All_samples/\n"
     ]
    }
   ],
   "source": [
    "def create_dir(name_dir):\n",
    "    if os.path.exists(name_dir):\n",
    "        shutil.rmtree(name_dir)\n",
    "    os.mkdir(name_dir)\n",
    "    return name_dir\n",
    "    \n",
    "All_samples_dir = create_dir(name_dir = '../All_samples/')\n",
    "print(All_samples_dir)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "88469bc1-f044-4620-b532-199b1ff1a8de",
   "metadata": {},
   "outputs": [],
   "source": [
    "def read_metadata_files(metadata_subdir, experiment_id): \n",
    "    \n",
    "    metadata_files = [f for f in listdir(metadata_subdir) if isfile(join(metadata_subdir, f))]\n",
    "    print('Files in metadata subdirectory:\\n{}\\n'.format(metadata_files))\n",
    "    \n",
    "    input_metadata_dir = list(filter(lambda x: ('_metadata_r' in x and experiment_id in x), metadata_files))[0]\n",
    "    metadata_df = pd.read_table(metadata_subdir + input_metadata_dir, delimiter = '\\t',  index_col = 0)\n",
    "    print('input_metadata_dir: {} - shape: {} \\n'.format(input_metadata_dir, metadata_df.shape))\n",
    "    return metadata_df\n",
    "\n",
    "\n",
    "metadata_df = read_metadata_files(metadata_subdir, experiment_id)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "87cff33e-ab3a-4ea5-bc06-1be4fa668b0b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "5f52f3f9-7a64-43c6-b3be-5d1a71b28849",
   "metadata": {},
   "source": [
    "# FeatureCounts for all narrowPeak (PCA plot)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "a2233deb-d249-440e-815d-334a6eec9bc8",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "sample_list: \n",
      "['SRR5876158', 'SRR5876159', 'SRR5876160', 'SRR5876161', 'SRR5876162', 'SRR5876163', 'SRR5876164', 'SRR5876165', 'SRR5876661', 'SRR5876662', 'SRR5876663', 'SRR5876664']\n",
      "\n",
      "narrowPeak_list: \n",
      "['/home/fuzan/fht.samba.data/experiments/ATACseq/NS-23.0061/alignment/macs2/SRR5876158_peaks.narrowPeak', '/home/fuzan/fht.samba.data/experiments/ATACseq/NS-23.0061/alignment/macs2/SRR5876159_peaks.narrowPeak', '/home/fuzan/fht.samba.data/experiments/ATACseq/NS-23.0061/alignment/macs2/SRR5876160_peaks.narrowPeak', '/home/fuzan/fht.samba.data/experiments/ATACseq/NS-23.0061/alignment/macs2/SRR5876161_peaks.narrowPeak', '/home/fuzan/fht.samba.data/experiments/ATACseq/NS-23.0061/alignment/macs2/SRR5876162_peaks.narrowPeak', '/home/fuzan/fht.samba.data/experiments/ATACseq/NS-23.0061/alignment/macs2/SRR5876163_peaks.narrowPeak', '/home/fuzan/fht.samba.data/experiments/ATACseq/NS-23.0061/alignment/macs2/SRR5876164_peaks.narrowPeak', '/home/fuzan/fht.samba.data/experiments/ATACseq/NS-23.0061/alignment/macs2/SRR5876165_peaks.narrowPeak', '/home/fuzan/fht.samba.data/experiments/ATACseq/NS-23.0061/alignment/macs2/SRR5876661_peaks.narrowPeak', '/home/fuzan/fht.samba.data/experiments/ATACseq/NS-23.0061/alignment/macs2/SRR5876662_peaks.narrowPeak', '/home/fuzan/fht.samba.data/experiments/ATACseq/NS-23.0061/alignment/macs2/SRR5876663_peaks.narrowPeak', '/home/fuzan/fht.samba.data/experiments/ATACseq/NS-23.0061/alignment/macs2/SRR5876664_peaks.narrowPeak']\n",
      "\n",
      "\n",
      "bam_list: \n",
      "['/home/fuzan/fht.samba.data/experiments/ATACseq/NS-23.0061/alignment/bam/SRR5876158.small.chrsorted.BAM', '/home/fuzan/fht.samba.data/experiments/ATACseq/NS-23.0061/alignment/bam/SRR5876159.small.chrsorted.BAM', '/home/fuzan/fht.samba.data/experiments/ATACseq/NS-23.0061/alignment/bam/SRR5876160.small.chrsorted.BAM', '/home/fuzan/fht.samba.data/experiments/ATACseq/NS-23.0061/alignment/bam/SRR5876161.small.chrsorted.BAM', '/home/fuzan/fht.samba.data/experiments/ATACseq/NS-23.0061/alignment/bam/SRR5876162.small.chrsorted.BAM', '/home/fuzan/fht.samba.data/experiments/ATACseq/NS-23.0061/alignment/bam/SRR5876163.small.chrsorted.BAM', '/home/fuzan/fht.samba.data/experiments/ATACseq/NS-23.0061/alignment/bam/SRR5876164.small.chrsorted.BAM', '/home/fuzan/fht.samba.data/experiments/ATACseq/NS-23.0061/alignment/bam/SRR5876165.small.chrsorted.BAM', '/home/fuzan/fht.samba.data/experiments/ATACseq/NS-23.0061/alignment/bam/SRR5876661.small.chrsorted.BAM', '/home/fuzan/fht.samba.data/experiments/ATACseq/NS-23.0061/alignment/bam/SRR5876662.small.chrsorted.BAM', '/home/fuzan/fht.samba.data/experiments/ATACseq/NS-23.0061/alignment/bam/SRR5876663.small.chrsorted.BAM', '/home/fuzan/fht.samba.data/experiments/ATACseq/NS-23.0061/alignment/bam/SRR5876664.small.chrsorted.BAM']\n",
      "\n",
      "\n",
      "       \u001b[44;37m ========== \u001b[0m\u001b[36m    _____ _    _ ____  _____  ______          _____  \n",
      "       \u001b[44;37m =====      \u001b[0m\u001b[36m   / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "       \u001b[44;37m   =====    \u001b[0m\u001b[36m  | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "       \u001b[44;37m     ====   \u001b[0m\u001b[36m   \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "       \u001b[44;37m       ==== \u001b[0m\u001b[36m   ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "       \u001b[44;37m ========== \u001b[0m\u001b[36m  |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\u001b[0m\n",
      "\t  v2.0.1\n",
      "\n",
      "//==========================\u001b[36m featureCounts setting \u001b[0m===========================\\\\\n",
      "||  \u001b[0m                                                                          ||\n",
      "||             Input files : \u001b[36m12 BAM files  \u001b[0m \u001b[0m                                  ||\n",
      "||                           \u001b[32mo\u001b[36m SRR5876158.small.chrsorted.BAM\u001b[0m \u001b[0m                ||\n",
      "||                           \u001b[32mo\u001b[36m SRR5876159.small.chrsorted.BAM\u001b[0m \u001b[0m                ||\n",
      "||                           \u001b[32mo\u001b[36m SRR5876160.small.chrsorted.BAM\u001b[0m \u001b[0m                ||\n",
      "||                           \u001b[32mo\u001b[36m SRR5876161.small.chrsorted.BAM\u001b[0m \u001b[0m                ||\n",
      "||                           \u001b[32mo\u001b[36m SRR5876162.small.chrsorted.BAM\u001b[0m \u001b[0m                ||\n",
      "||                           \u001b[32mo\u001b[36m SRR5876163.small.chrsorted.BAM\u001b[0m \u001b[0m                ||\n",
      "||                           \u001b[32mo\u001b[36m SRR5876164.small.chrsorted.BAM\u001b[0m \u001b[0m                ||\n",
      "||                           \u001b[32mo\u001b[36m SRR5876165.small.chrsorted.BAM\u001b[0m \u001b[0m                ||\n",
      "||                           \u001b[32mo\u001b[36m SRR5876661.small.chrsorted.BAM\u001b[0m \u001b[0m                ||\n",
      "||                           \u001b[32mo\u001b[36m SRR5876662.small.chrsorted.BAM\u001b[0m \u001b[0m                ||\n",
      "||                           \u001b[32mo\u001b[36m SRR5876663.small.chrsorted.BAM\u001b[0m \u001b[0m                ||\n",
      "||                           \u001b[32mo\u001b[36m SRR5876664.small.chrsorted.BAM\u001b[0m \u001b[0m                ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "||             Output file : \u001b[36mAll_samples.orig.txt\u001b[0m \u001b[0m                            ||\n",
      "||                 Summary : \u001b[36mAll_samples.orig.txt.summary\u001b[0m \u001b[0m                    ||\n",
      "||              Annotation : \u001b[36mAll_samples.merged.saf (SAF)\u001b[0m \u001b[0m                    ||\n",
      "||      Dir for temp files : \u001b[36m../All_samples\u001b[0m \u001b[0m                                  ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "||                 Threads : \u001b[36m1\u001b[0m \u001b[0m                                               ||\n",
      "||                   Level : \u001b[36mmeta-feature level\u001b[0m \u001b[0m                              ||\n",
      "||              Paired-end : \u001b[36mno\u001b[0m \u001b[0m                                              ||\n",
      "||      Multimapping reads : \u001b[36mnot counted\u001b[0m \u001b[0m                                     ||\n",
      "|| Multi-overlapping reads : \u001b[36mnot counted\u001b[0m \u001b[0m                                     ||\n",
      "||   Min overlapping bases : \u001b[36m1\u001b[0m \u001b[0m                                               ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "//=================================\u001b[36m Running \u001b[0m==================================\\\\\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Load annotation file All_samples.merged.saf \u001b[0m... \u001b[0m                           ||\n",
      "||    Features : \u001b[36m1496390\u001b[0m \u001b[0m                                                     ||\n",
      "||    Meta-features : \u001b[36m1496390\u001b[0m \u001b[0m                                                ||\n",
      "||    Chromosomes/contigs : \u001b[36m24\u001b[0m \u001b[0m                                               ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file SRR5876158.small.chrsorted.BAM... \u001b[0m                        ||\n",
      "||    WARNING: \u001b[36mPaired-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m9598520\u001b[0m \u001b[0m                                             ||\n",
      "||    Successfully assigned alignments : \u001b[36m4460873 (46.5%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m0.61 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file SRR5876159.small.chrsorted.BAM... \u001b[0m                        ||\n",
      "||    WARNING: \u001b[36mPaired-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m6631432\u001b[0m \u001b[0m                                             ||\n",
      "||    Successfully assigned alignments : \u001b[36m4040734 (60.9%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m0.37 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file SRR5876160.small.chrsorted.BAM... \u001b[0m                        ||\n",
      "||    WARNING: \u001b[36mPaired-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m15085372\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m7869906 (52.2%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m0.89 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file SRR5876161.small.chrsorted.BAM... \u001b[0m                        ||\n",
      "||    WARNING: \u001b[36mPaired-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m5747942\u001b[0m \u001b[0m                                             ||\n",
      "||    Successfully assigned alignments : \u001b[36m3456419 (60.1%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m0.31 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file SRR5876162.small.chrsorted.BAM... \u001b[0m                        ||\n",
      "||    WARNING: \u001b[36mPaired-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m13048224\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m6347494 (48.6%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m0.83 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file SRR5876163.small.chrsorted.BAM... \u001b[0m                        ||\n",
      "||    WARNING: \u001b[36mPaired-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m7156354\u001b[0m \u001b[0m                                             ||\n",
      "||    Successfully assigned alignments : \u001b[36m4141159 (57.9%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m0.40 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file SRR5876164.small.chrsorted.BAM... \u001b[0m                        ||\n",
      "||    WARNING: \u001b[36mPaired-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m12931920\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m6137472 (47.5%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m0.84 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file SRR5876165.small.chrsorted.BAM... \u001b[0m                        ||\n",
      "||    WARNING: \u001b[36mPaired-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m5822936\u001b[0m \u001b[0m                                             ||\n",
      "||    Successfully assigned alignments : \u001b[36m3021527 (51.9%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m0.36 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file SRR5876661.small.chrsorted.BAM... \u001b[0m                        ||\n",
      "||    WARNING: \u001b[36mPaired-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m10292336\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m4320131 (42.0%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m0.66 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file SRR5876662.small.chrsorted.BAM... \u001b[0m                        ||\n",
      "||    WARNING: \u001b[36mPaired-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m28519554\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m8943865 (31.4%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m1.86 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file SRR5876663.small.chrsorted.BAM... \u001b[0m                        ||\n",
      "||    WARNING: \u001b[36mPaired-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m14775364\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m6317734 (42.8%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m0.96 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file SRR5876664.small.chrsorted.BAM... \u001b[0m                        ||\n",
      "||    WARNING: \u001b[36mPaired-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m27175488\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m8855023 (32.6%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m1.77 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Write the final count table. \u001b[0m                                              ||\n",
      "|| Write the read assignment summary. \u001b[0m                                        ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Summary of counting results can be found in file \"../All_samples/All_samp \u001b[0m ||\n",
      "|| les.orig.txt.summary\" \u001b[0m                                                     ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "peak_id\tchr\tstart\tend\tSRR5876158\tSRR5876159\tSRR5876160\tSRR5876161\tSRR5876162\tSRR5876163\tSRR5876164\tSRR5876165\tSRR5876661\tSRR5876662\tSRR5876663\tSRR5876664\n",
      "chr1:41047-41197\tchr1\t41048\t41197\t0\t0\t0\t0\t0\t0\t4\t0\t0\t2\t3\t0\n",
      "chr1:41539-41700\tchr1\t41540\t41700\t6\t0\t0\t0\t0\t0\t0\t0\t0\t6\t4\t0\n",
      "chr1:42249-42399\tchr1\t42250\t42399\t0\t0\t0\t0\t0\t0\t2\t2\t4\t0\t0\t5\n",
      "chr1:50251-50459\tchr1\t50252\t50459\t6\t0\t2\t0\t2\t0\t2\t0\t2\t4\t0\t4\n",
      "chr1:51266-51416\tchr1\t51267\t51416\t0\t0\t1\t0\t0\t0\t2\t0\t0\t0\t6\t0\n",
      "chr1:52762-52969\tchr1\t52763\t52969\t0\t0\t6\t0\t0\t1\t0\t1\t0\t4\t2\t2\n",
      "chr1:54941-55641\tchr1\t54942\t55641\t0\t0\t2\t0\t4\t0\t10\t4\t3\t6\t10\t4\n",
      "chr1:56325-56487\tchr1\t56326\t56487\t2\t0\t6\t0\t2\t0\t0\t0\t2\t0\t0\t2\n",
      "chr1:57202-57549\tchr1\t57203\t57549\t0\t6\t2\t0\t2\t0\t2\t0\t0\t2\t0\t0\n"
     ]
    }
   ],
   "source": [
    "def all_sample_featureCounts(metadata_df):\n",
    "    sample_list = list(metadata_df.index)\n",
    "    print(\"sample_list: \\n{}\\n\".format(sample_list))\n",
    "\n",
    "    # 1) filter chromosomes that are not part of the chromosome set (should be added to previous step) and combined test+grp and negative_cntrl_grp\n",
    "    # write path of each narrowPeak file\n",
    "    narrowPeak_list = [PEAKS_dir + group + '_peaks.narrowPeak' for group in sample_list]\n",
    "    print(\"narrowPeak_list: \\n{}\\n\".format(narrowPeak_list))\n",
    "    for narrowPeak in narrowPeak_list:\n",
    "        cmd_cat = \"\"\"grep -h '^chr' {narrowPeak} >> {All_samples_dir}All_samples.narrowPeak\"\"\".format(narrowPeak=narrowPeak, All_samples_dir=All_samples_dir)\n",
    "        !{cmd_cat}\n",
    "\n",
    "    # 2) sort by chromosome positions (chr, start)\n",
    "    !sort -k1,1V -k2,2n -k3,3n {All_samples_dir}All_samples.narrowPeak > {All_samples_dir}All_samples.sorted.narrowPeak\n",
    "\n",
    "    # 3) bedtools merge\n",
    "    !bedtools merge -d 50 -i  {All_samples_dir}All_samples.sorted.narrowPeak > {All_samples_dir}All_samples.merged.narrowPeak\n",
    "\n",
    "    # 6) add header and convert merged.bed file to saf file for each sample\n",
    "    cmd_saf = \"\"\"awk 'BEGIN{{FS=OFS=\"\\t\"; print \"GeneID\\tchr\\tstart\\tend\\tstrand\"}} {{print $1\":\"$2\"-\"$3, $1, $2+1, $3, \".\"}}'  {All_samples_dir}All_samples.merged.narrowPeak > {All_samples_dir}All_samples.merged.saf\"\"\".format(All_samples_dir=All_samples_dir)\n",
    "    ! {cmd_saf}\n",
    "\n",
    "    # 7) Call featureCounts on saf file for each sample\n",
    "    bam_list = [BAM_DIR + group + '.small.chrsorted.BAM' for group in sample_list]\n",
    "    print(\"\\nbam_list: \\n{}\\n\".format(bam_list))\n",
    "    bam_list_bash = '\\t'.join(bam_list)\n",
    "    !featureCounts -T 48 -a {All_samples_dir}All_samples.merged.saf -F SAF  -o {All_samples_dir}All_samples.orig.txt {bam_list_bash}\n",
    "\n",
    "    # 8) Remove header and reformat header for DPA logFC computation\n",
    "    !sed -i '1,2d' {All_samples_dir}All_samples.orig.txt\n",
    "    \n",
    "    # 9) reformat header for DPA logFC computation\n",
    "    sample_list_bash = '\\t'.join(sample_list)\n",
    "    sample_number = [1,2,3,4] + list(np.arange(7, 7+ len(sample_list), 1))\n",
    "    sample_number_str = [str(i) for i in sample_number]\n",
    "    sample_number_bash = '$' + ', $'.join(sample_number_str)\n",
    "    cmd_header = \"\"\"awk 'BEGIN{{FS=OFS=\"\\t\"; print \"peak_id\\tchr\\tstart\\tend\\t{sample_list_bash}\"}} {{print {sample_number_bash}}}' {All_samples_dir}All_samples.orig.txt > {All_samples_dir}All_samples.txt\"\"\".format(sample_number_bash=sample_number_bash, sample_list_bash=sample_list_bash, All_samples_dir=All_samples_dir)\n",
    "    ! {cmd_header}\n",
    "    \n",
    "    !head {All_samples_dir}All_samples.txt\n",
    "\n",
    "all_sample_featureCounts(metadata_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "356f0793-9b8f-47c7-b60a-67cc5ee2d2e7",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
