{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "c077e4b4-0589-446d-84db-1b85e37bc1f0",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import shutil\n",
    "\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "\n",
    "from os import listdir\n",
    "from os.path import isfile, join\n",
    "import json \n",
    "\n",
    "import plotly\n",
    "import plotly.express as px\n",
    "import plotly.io as pio"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a0b05a14-44a2-432e-953a-1076176e32d7",
   "metadata": {},
   "source": [
    "# featureCounts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "563e2b64-2d8b-4390-b435-c3e456ad55c0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "metadata_subdir exists:True\n",
      "\n",
      "PEAKS_dir exists:True\n",
      "\n",
      "BAM_DIR exists:True\n",
      "\n",
      "idr_bed_dir exists:True\n",
      "\n"
     ]
    }
   ],
   "source": [
    "experiment_id = \"NS-23.0061\"\n",
    "\n",
    "IDR = \"nIDR\" # \"standard\"  \n",
    "\n",
    "metadata_subdir = os.path.expanduser(\"~/fht.samba.data/experiments/ATACseq/{}/analysis/metadata/\".format(experiment_id))\n",
    "print('metadata_subdir exists:{}\\n'.format(os.path.exists(metadata_subdir)))\n",
    "\n",
    "PEAKS_dir = os.path.expanduser(\"~/fht.samba.data/experiments/ATACseq/{}/alignment/macs2/\".format(experiment_id))\n",
    "print('PEAKS_dir exists:{}\\n'.format(os.path.exists(PEAKS_dir)))\n",
    "\n",
    "BAM_DIR = os.path.expanduser(\"~/fht.samba.data/experiments/ATACseq/{}/alignment/bam/\".format(experiment_id))\n",
    "print('BAM_DIR exists:{}\\n'.format(os.path.exists(BAM_DIR)))\n",
    "\n",
    "idr_bed_dir = \"../idr_bed/\"\n",
    "print('idr_bed_dir exists:{}\\n'.format(os.path.exists(idr_bed_dir)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "f4501abc-0331-4d2d-8bdd-75f9835d96d4",
   "metadata": {},
   "outputs": [],
   "source": [
    "def create_dir(name_dir):\n",
    "    if os.path.exists(name_dir):\n",
    "        shutil.rmtree(name_dir)\n",
    "    os.mkdir(name_dir)\n",
    "    return name_dir"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "94bedf71-5368-4923-8f68-5006de295c66",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "../merged_IDR_narrowPeak/\n",
      "../featureCounts/\n"
     ]
    }
   ],
   "source": [
    "merged_IDR_narrowPeak_dir = create_dir(\"../merged_IDR_narrowPeak/\")\n",
    "print(merged_IDR_narrowPeak_dir)\n",
    "\n",
    "featureCounts_dir = create_dir(\"../featureCounts/\")\n",
    "print(featureCounts_dir)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "3bef5334-af36-492a-a36e-6d520b0a956c",
   "metadata": {},
   "outputs": [],
   "source": [
    "if IDR == \"standard\":\n",
    "    IDR_narrowPEAK_dir = create_dir(\"../IDR_narrowPEAK/\")\n",
    "    print(IDR_narrowPEAK_dir)\n",
    "    \n",
    "    macs2_dir = \"~/fht.samba.data/experiments/ATACseq/{}/alignment/macs2/*IDR0.1.narrowPeak.gz\".format(experiment_id) \n",
    "    \n",
    "    !cp {macs2_dir} {IDR_narrowPEAK_dir}\n",
    "    !!gunzip {IDR_narrowPEAK_dir}/*.gz*"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "195613a3-0249-4a71-a058-906f6c3a851c",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "output_filename : NS-23.0061_contrast_dict.json\n",
      "output_filepath : /home/fuzan/fht.samba.data/experiments/ATACseq/NS-23.0061/analysis/metadata/NS-23.0061_contrast_dict.json\n"
     ]
    }
   ],
   "source": [
    "def read_contrast_dict(metadata_subdir, experiment_id): \n",
    "    output_filename = experiment_id + '_contrast_dict.json'\n",
    "    print(\"output_filename : {}\".format(output_filename))\n",
    "\n",
    "    output_filepath = os.path.join(metadata_subdir, output_filename)\n",
    "    print(\"output_filepath : {}\".format(output_filepath))\n",
    "    \n",
    "    # Opening JSON file\n",
    "    with open(output_filepath) as json_file:\n",
    "        contrast_dict = json.load(json_file)\n",
    "        \n",
    "    return contrast_dict\n",
    "\n",
    "contrast_dict = read_contrast_dict(metadata_subdir, experiment_id)\n",
    "# contrast_dict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "cd46f664-44b9-446e-8a1d-3e2eaa3c9dca",
   "metadata": {},
   "outputs": [],
   "source": [
    "# contrast_dict = dict(filter(lambda item:('KMS20' in item[0]) or ('KO52' in item[0]), contrast_dict.items()))\n",
    "# contrast_dict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "18cbf3c5-c217-44a5-8150-94208d052eac",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "nIDR version\n",
      "\n",
      "contrast_list: \n",
      "['ARID1AKO_HCT116_WT_HCT116', 'ARID1BKD_HCT116_WT_HCT116', 'ARID1AKO_ARID1BKD_HCT116_WT_HCT116', 'ARID1BKD_TOV21G_WT_TOV21G', 'ARID1AKO_ARID1BKD_HCT116_ARID1AKO_HCT116']\n",
      "\n",
      "contrast: ARID1BKD_HCT116_WT_HCT116\n",
      "test_group: ARID1BKD_HCT116 -> ['SRR5876160', 'SRR5876161']\n",
      "sample_negative_ctrl_grp: WT_HCT116 -> ['SRR5876158', 'SRR5876159']\n",
      "\n",
      "contrast: ARID1AKO_HCT116_WT_HCT116\n",
      "test_group: ARID1AKO_HCT116 -> ['SRR5876162', 'SRR5876163']\n",
      "sample_negative_ctrl_grp: WT_HCT116 -> ['SRR5876158', 'SRR5876159']\n",
      "\n",
      "contrast: ARID1AKO_ARID1BKD_HCT116_WT_HCT116\n",
      "test_group: ARID1AKO_ARID1BKD_HCT116 -> ['SRR5876164', 'SRR5876165']\n",
      "sample_negative_ctrl_grp: WT_HCT116 -> ['SRR5876158', 'SRR5876159']\n",
      "\n",
      "contrast: ARID1AKO_ARID1BKD_HCT116_ARID1AKO_HCT116\n",
      "test_group: ARID1AKO_ARID1BKD_HCT116 -> ['SRR5876164', 'SRR5876165']\n",
      "sample_negative_ctrl_grp: ARID1AKO_HCT116 -> ['SRR5876162', 'SRR5876163']\n",
      "\n",
      "contrast: ARID1BKD_TOV21G_WT_TOV21G\n",
      "test_group: ARID1BKD_TOV21G -> ['SRR5876663', 'SRR5876664']\n",
      "sample_negative_ctrl_grp: WT_TOV21G -> ['SRR5876661', 'SRR5876662']\n",
      "\n"
     ]
    }
   ],
   "source": [
    "def merge_narrowPeak(contrast_dict, IDR):\n",
    "    if IDR == 'standard':\n",
    "        print('Standard IDR version')\n",
    "    if IDR == 'nIDR':\n",
    "        print('nIDR version')        \n",
    "\n",
    "    contrast_list = list(set(contrast_dict.keys()))\n",
    "    print(\"\\ncontrast_list: \\n{}\\n\".format(contrast_list))\n",
    "\n",
    "    for contrast in list(contrast_dict.keys()):\n",
    "            print(\"contrast: {}\".format(contrast))\n",
    "\n",
    "            # # 1) combine test+grp and negative_cntrl_grp\n",
    "            # if IDR == 'standard':\n",
    "            #     test_group_peak_file = contrasts_df.loc[list(contrasts_df[contrasts_df['file_friendly_name'] == contrast].index)[0], 'test_peak_file']\n",
    "            #     negative_ctrl_grp_peak_file = contrasts_df.loc[list(contrasts_df[contrasts_df['file_friendly_name'] == contrast].index)[0], 'negative_ctrl_peak_file']\n",
    "            #     print(test_group_peak_file)\n",
    "            #     print(negative_ctrl_grp_peak_file)\n",
    "            #     !grep -h '^chr'  {IDR_narrowPEAK_dir}{test_group_peak_file} {IDR_narrowPEAK_dir}{negative_ctrl_grp_peak_file} > {merged_IDR_narrowPeak_dir}{contrast}.combined.narrowPeak\n",
    "\n",
    "            if IDR == 'nIDR': \n",
    "                test_group = list(contrast_dict[contrast]['test_group'].keys())[0]\n",
    "                sample_test_group = list(contrast_dict[contrast]['test_group'].values())[0]\n",
    "                print('test_group: {} -> {}'.format(test_group, sample_test_group))\n",
    "\n",
    "                negative_ctrl_grp = list(contrast_dict[contrast]['negative_ctrl_grp'].keys())[0]\n",
    "                sample_negative_ctrl_grp = list(contrast_dict[contrast]['negative_ctrl_grp'].values())[0]\n",
    "                print(\"sample_negative_ctrl_grp: {} -> {}\\n\".format(negative_ctrl_grp, sample_negative_ctrl_grp))\n",
    "                \n",
    "                cmd_cat = \"\"\"grep -h '^chr'  {idr_bed_dir}{test_group}.IDR.narrowPeak {idr_bed_dir}{negative_ctrl_grp}.IDR.narrowPeak > {merged_IDR_narrowPeak_dir}{contrast}.combined.narrowPeak\"\"\".format(idr_bed_dir=idr_bed_dir, test_group=test_group, negative_ctrl_grp=negative_ctrl_grp, contrast=contrast, merged_IDR_narrowPeak_dir=merged_IDR_narrowPeak_dir)\n",
    "                # print(cmd_cat)\n",
    "                ! {cmd_cat}\n",
    "\n",
    "            # 2) sort by chromosome positions (chr, start)\n",
    "            cmd_sort = \"\"\"sort -k1,1V -k2,2n -k3,3n {merged_IDR_narrowPeak_dir}{contrast}.combined.narrowPeak > {merged_IDR_narrowPeak_dir}{contrast}.sorted.narrowPeak\"\"\".format(merged_IDR_narrowPeak_dir=merged_IDR_narrowPeak_dir, contrast=contrast)\n",
    "            # print(cmd_sort)\n",
    "            ! {cmd_sort}\n",
    "            \n",
    "            # 3) bedtools merge\n",
    "            cmd_merge = \"\"\"bedtools merge -d 50 -i  {merged_IDR_narrowPeak_dir}{contrast}.sorted.narrowPeak > {merged_IDR_narrowPeak_dir}{contrast}.merged.narrowPeak\"\"\".format(merged_IDR_narrowPeak_dir=merged_IDR_narrowPeak_dir, contrast=contrast)\n",
    "            # print(cmd_merge)\n",
    "            !{cmd_merge}\n",
    "            \n",
    "            # 6) add header and convert merged.bed file to saf file for each contrast\n",
    "            cmd_saf = \"\"\"awk 'BEGIN{{FS=OFS=\"\\t\"; print \"GeneID\\tchr\\tstart\\tend\\tstrand\"}} {{print $1\":\"$2\"-\"$3, $1, $2+1, $3, \".\"}}'  {merged_IDR_narrowPeak_dir}{contrast}.merged.narrowPeak > {merged_IDR_narrowPeak_dir}{contrast}.merged.saf\"\"\".format(merged_IDR_narrowPeak_dir=merged_IDR_narrowPeak_dir, contrast=contrast)\n",
    "            # print(cmd_saf)\n",
    "            ! {cmd_saf}\n",
    "\n",
    "            \n",
    "merge_narrowPeak(contrast_dict, IDR)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "044f92d4-3692-47d4-8018-e4779ed22ec7",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "contrast_list: \n",
      "['ARID1AKO_HCT116_WT_HCT116', 'ARID1BKD_HCT116_WT_HCT116', 'ARID1AKO_ARID1BKD_HCT116_WT_HCT116', 'ARID1BKD_TOV21G_WT_TOV21G', 'ARID1AKO_ARID1BKD_HCT116_ARID1AKO_HCT116']\n",
      "\n",
      "contrast: ARID1BKD_HCT116_WT_HCT116\n",
      "test_group: ARID1BKD_HCT116 -> ['SRR5876160', 'SRR5876161']\n",
      "sample_negative_ctrl_grp: WT_HCT116 -> ['SRR5876158', 'SRR5876159']\n",
      "\n",
      "\n",
      "       \u001b[44;37m ========== \u001b[0m\u001b[36m    _____ _    _ ____  _____  ______          _____  \n",
      "       \u001b[44;37m =====      \u001b[0m\u001b[36m   / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "       \u001b[44;37m   =====    \u001b[0m\u001b[36m  | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "       \u001b[44;37m     ====   \u001b[0m\u001b[36m   \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "       \u001b[44;37m       ==== \u001b[0m\u001b[36m   ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "       \u001b[44;37m ========== \u001b[0m\u001b[36m  |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\u001b[0m\n",
      "\t  v2.0.1\n",
      "\n",
      "//==========================\u001b[36m featureCounts setting \u001b[0m===========================\\\\\n",
      "||  \u001b[0m                                                                          ||\n",
      "||             Input files : \u001b[36m4 BAM files  \u001b[0m \u001b[0m                                   ||\n",
      "||                           \u001b[32mo\u001b[36m SRR5876160.small.chrsorted.BAM\u001b[0m \u001b[0m                ||\n",
      "||                           \u001b[32mo\u001b[36m SRR5876161.small.chrsorted.BAM\u001b[0m \u001b[0m                ||\n",
      "||                           \u001b[32mo\u001b[36m SRR5876158.small.chrsorted.BAM\u001b[0m \u001b[0m                ||\n",
      "||                           \u001b[32mo\u001b[36m SRR5876159.small.chrsorted.BAM\u001b[0m \u001b[0m                ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "||             Output file : \u001b[36mARID1BKD_HCT116_WT_HCT116.orig.txt\u001b[0m \u001b[0m              ||\n",
      "||                 Summary : \u001b[36mARID1BKD_HCT116_WT_HCT116.orig.txt.summary\u001b[0m \u001b[0m      ||\n",
      "||              Annotation : \u001b[36mARID1BKD_HCT116_WT_HCT116.merged.saf (SAF)\u001b[0m \u001b[0m      ||\n",
      "||      Dir for temp files : \u001b[36m../merged_IDR_narrowPeak\u001b[0m \u001b[0m                        ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "||                 Threads : \u001b[36m1\u001b[0m \u001b[0m                                               ||\n",
      "||                   Level : \u001b[36mmeta-feature level\u001b[0m \u001b[0m                              ||\n",
      "||              Paired-end : \u001b[36mno\u001b[0m \u001b[0m                                              ||\n",
      "||      Multimapping reads : \u001b[36mnot counted\u001b[0m \u001b[0m                                     ||\n",
      "|| Multi-overlapping reads : \u001b[36mnot counted\u001b[0m \u001b[0m                                     ||\n",
      "||   Min overlapping bases : \u001b[36m1\u001b[0m \u001b[0m                                               ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "//=================================\u001b[36m Running \u001b[0m==================================\\\\\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Load annotation file ARID1BKD_HCT116_WT_HCT116.merged.saf \u001b[0m... \u001b[0m             ||\n",
      "||    Features : \u001b[36m76523\u001b[0m \u001b[0m                                                       ||\n",
      "||    Meta-features : \u001b[36m76523\u001b[0m \u001b[0m                                                  ||\n",
      "||    Chromosomes/contigs : \u001b[36m24\u001b[0m \u001b[0m                                               ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file SRR5876160.small.chrsorted.BAM... \u001b[0m                        ||\n",
      "||    WARNING: \u001b[36mPaired-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m15085372\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m4507558 (29.9%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m0.87 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file SRR5876161.small.chrsorted.BAM... \u001b[0m                        ||\n",
      "||    WARNING: \u001b[36mPaired-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m5747942\u001b[0m \u001b[0m                                             ||\n",
      "||    Successfully assigned alignments : \u001b[36m2485310 (43.2%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m0.32 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file SRR5876158.small.chrsorted.BAM... \u001b[0m                        ||\n",
      "||    WARNING: \u001b[36mPaired-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m9598520\u001b[0m \u001b[0m                                             ||\n",
      "||    Successfully assigned alignments : \u001b[36m2486692 (25.9%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m0.60 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file SRR5876159.small.chrsorted.BAM... \u001b[0m                        ||\n",
      "||    WARNING: \u001b[36mPaired-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m6631432\u001b[0m \u001b[0m                                             ||\n",
      "||    Successfully assigned alignments : \u001b[36m2811491 (42.4%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m0.37 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Write the final count table. \u001b[0m                                              ||\n",
      "|| Write the read assignment summary. \u001b[0m                                        ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Summary of counting results can be found in file \"../merged_IDR_narrowPea \u001b[0m ||\n",
      "|| k/ARID1BKD_HCT116_WT_HCT116.orig.txt.summary\" \u001b[0m                             ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "contrast: ARID1AKO_HCT116_WT_HCT116\n",
      "test_group: ARID1AKO_HCT116 -> ['SRR5876162', 'SRR5876163']\n",
      "sample_negative_ctrl_grp: WT_HCT116 -> ['SRR5876158', 'SRR5876159']\n",
      "\n",
      "\n",
      "       \u001b[44;37m ========== \u001b[0m\u001b[36m    _____ _    _ ____  _____  ______          _____  \n",
      "       \u001b[44;37m =====      \u001b[0m\u001b[36m   / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "       \u001b[44;37m   =====    \u001b[0m\u001b[36m  | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "       \u001b[44;37m     ====   \u001b[0m\u001b[36m   \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "       \u001b[44;37m       ==== \u001b[0m\u001b[36m   ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "       \u001b[44;37m ========== \u001b[0m\u001b[36m  |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\u001b[0m\n",
      "\t  v2.0.1\n",
      "\n",
      "//==========================\u001b[36m featureCounts setting \u001b[0m===========================\\\\\n",
      "||  \u001b[0m                                                                          ||\n",
      "||             Input files : \u001b[36m4 BAM files  \u001b[0m \u001b[0m                                   ||\n",
      "||                           \u001b[32mo\u001b[36m SRR5876162.small.chrsorted.BAM\u001b[0m \u001b[0m                ||\n",
      "||                           \u001b[32mo\u001b[36m SRR5876163.small.chrsorted.BAM\u001b[0m \u001b[0m                ||\n",
      "||                           \u001b[32mo\u001b[36m SRR5876158.small.chrsorted.BAM\u001b[0m \u001b[0m                ||\n",
      "||                           \u001b[32mo\u001b[36m SRR5876159.small.chrsorted.BAM\u001b[0m \u001b[0m                ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "||             Output file : \u001b[36mARID1AKO_HCT116_WT_HCT116.orig.txt\u001b[0m \u001b[0m              ||\n",
      "||                 Summary : \u001b[36mARID1AKO_HCT116_WT_HCT116.orig.txt.summary\u001b[0m \u001b[0m      ||\n",
      "||              Annotation : \u001b[36mARID1AKO_HCT116_WT_HCT116.merged.saf (SAF)\u001b[0m \u001b[0m      ||\n",
      "||      Dir for temp files : \u001b[36m../merged_IDR_narrowPeak\u001b[0m \u001b[0m                        ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "||                 Threads : \u001b[36m1\u001b[0m \u001b[0m                                               ||\n",
      "||                   Level : \u001b[36mmeta-feature level\u001b[0m \u001b[0m                              ||\n",
      "||              Paired-end : \u001b[36mno\u001b[0m \u001b[0m                                              ||\n",
      "||      Multimapping reads : \u001b[36mnot counted\u001b[0m \u001b[0m                                     ||\n",
      "|| Multi-overlapping reads : \u001b[36mnot counted\u001b[0m \u001b[0m                                     ||\n",
      "||   Min overlapping bases : \u001b[36m1\u001b[0m \u001b[0m                                               ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "//=================================\u001b[36m Running \u001b[0m==================================\\\\\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Load annotation file ARID1AKO_HCT116_WT_HCT116.merged.saf \u001b[0m... \u001b[0m             ||\n",
      "||    Features : \u001b[36m74739\u001b[0m \u001b[0m                                                       ||\n",
      "||    Meta-features : \u001b[36m74739\u001b[0m \u001b[0m                                                  ||\n",
      "||    Chromosomes/contigs : \u001b[36m24\u001b[0m \u001b[0m                                               ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file SRR5876162.small.chrsorted.BAM... \u001b[0m                        ||\n",
      "||    WARNING: \u001b[36mPaired-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m13048224\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m3381393 (25.9%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m0.81 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file SRR5876163.small.chrsorted.BAM... \u001b[0m                        ||\n",
      "||    WARNING: \u001b[36mPaired-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m7156354\u001b[0m \u001b[0m                                             ||\n",
      "||    Successfully assigned alignments : \u001b[36m2648734 (37.0%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m0.40 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file SRR5876158.small.chrsorted.BAM... \u001b[0m                        ||\n",
      "||    WARNING: \u001b[36mPaired-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m9598520\u001b[0m \u001b[0m                                             ||\n",
      "||    Successfully assigned alignments : \u001b[36m2474072 (25.8%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m0.59 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file SRR5876159.small.chrsorted.BAM... \u001b[0m                        ||\n",
      "||    WARNING: \u001b[36mPaired-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m6631432\u001b[0m \u001b[0m                                             ||\n",
      "||    Successfully assigned alignments : \u001b[36m2788677 (42.1%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m0.37 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Write the final count table. \u001b[0m                                              ||\n",
      "|| Write the read assignment summary. \u001b[0m                                        ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Summary of counting results can be found in file \"../merged_IDR_narrowPea \u001b[0m ||\n",
      "|| k/ARID1AKO_HCT116_WT_HCT116.orig.txt.summary\" \u001b[0m                             ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "contrast: ARID1AKO_ARID1BKD_HCT116_WT_HCT116\n",
      "test_group: ARID1AKO_ARID1BKD_HCT116 -> ['SRR5876164', 'SRR5876165']\n",
      "sample_negative_ctrl_grp: WT_HCT116 -> ['SRR5876158', 'SRR5876159']\n",
      "\n",
      "\n",
      "       \u001b[44;37m ========== \u001b[0m\u001b[36m    _____ _    _ ____  _____  ______          _____  \n",
      "       \u001b[44;37m =====      \u001b[0m\u001b[36m   / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "       \u001b[44;37m   =====    \u001b[0m\u001b[36m  | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "       \u001b[44;37m     ====   \u001b[0m\u001b[36m   \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "       \u001b[44;37m       ==== \u001b[0m\u001b[36m   ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "       \u001b[44;37m ========== \u001b[0m\u001b[36m  |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\u001b[0m\n",
      "\t  v2.0.1\n",
      "\n",
      "//==========================\u001b[36m featureCounts setting \u001b[0m===========================\\\\\n",
      "||  \u001b[0m                                                                          ||\n",
      "||             Input files : \u001b[36m4 BAM files  \u001b[0m \u001b[0m                                   ||\n",
      "||                           \u001b[32mo\u001b[36m SRR5876164.small.chrsorted.BAM\u001b[0m \u001b[0m                ||\n",
      "||                           \u001b[32mo\u001b[36m SRR5876165.small.chrsorted.BAM\u001b[0m \u001b[0m                ||\n",
      "||                           \u001b[32mo\u001b[36m SRR5876158.small.chrsorted.BAM\u001b[0m \u001b[0m                ||\n",
      "||                           \u001b[32mo\u001b[36m SRR5876159.small.chrsorted.BAM\u001b[0m \u001b[0m                ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "||             Output file : \u001b[36mARID1AKO_ARID1BKD_HCT116_WT_HCT116.orig.txt\u001b[0m \u001b[0m     ||\n",
      "||                 Summary : \u001b[36mARID1AKO_ARID1BKD_HCT116_WT_HCT116.orig.txt.\u001b[0m ...\u001b[0m \u001b[0m||\n",
      "||              Annotation : \u001b[36mARID1AKO_ARID1BKD_HCT116_WT_HCT116.merged.sa\u001b[0m ...\u001b[0m \u001b[0m||\n",
      "||      Dir for temp files : \u001b[36m../merged_IDR_narrowPeak\u001b[0m \u001b[0m                        ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "||                 Threads : \u001b[36m1\u001b[0m \u001b[0m                                               ||\n",
      "||                   Level : \u001b[36mmeta-feature level\u001b[0m \u001b[0m                              ||\n",
      "||              Paired-end : \u001b[36mno\u001b[0m \u001b[0m                                              ||\n",
      "||      Multimapping reads : \u001b[36mnot counted\u001b[0m \u001b[0m                                     ||\n",
      "|| Multi-overlapping reads : \u001b[36mnot counted\u001b[0m \u001b[0m                                     ||\n",
      "||   Min overlapping bases : \u001b[36m1\u001b[0m \u001b[0m                                               ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "//=================================\u001b[36m Running \u001b[0m==================================\\\\\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Load annotation file ARID1AKO_ARID1BKD_HCT116_WT_HCT116.merged.saf \u001b[0m... \u001b[0m    ||\n",
      "||    Features : \u001b[36m75906\u001b[0m \u001b[0m                                                       ||\n",
      "||    Meta-features : \u001b[36m75906\u001b[0m \u001b[0m                                                  ||\n",
      "||    Chromosomes/contigs : \u001b[36m24\u001b[0m \u001b[0m                                               ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file SRR5876164.small.chrsorted.BAM... \u001b[0m                        ||\n",
      "||    WARNING: \u001b[36mPaired-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m12931920\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m3192150 (24.7%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m0.83 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file SRR5876165.small.chrsorted.BAM... \u001b[0m                        ||\n",
      "||    WARNING: \u001b[36mPaired-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m5822936\u001b[0m \u001b[0m                                             ||\n",
      "||    Successfully assigned alignments : \u001b[36m1748466 (30.0%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m0.33 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file SRR5876158.small.chrsorted.BAM... \u001b[0m                        ||\n",
      "||    WARNING: \u001b[36mPaired-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m9598520\u001b[0m \u001b[0m                                             ||\n",
      "||    Successfully assigned alignments : \u001b[36m2472457 (25.8%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m0.62 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file SRR5876159.small.chrsorted.BAM... \u001b[0m                        ||\n",
      "||    WARNING: \u001b[36mPaired-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m6631432\u001b[0m \u001b[0m                                             ||\n",
      "||    Successfully assigned alignments : \u001b[36m2779549 (41.9%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m0.36 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Write the final count table. \u001b[0m                                              ||\n",
      "|| Write the read assignment summary. \u001b[0m                                        ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Summary of counting results can be found in file \"../merged_IDR_narrowPea \u001b[0m ||\n",
      "|| k/ARID1AKO_ARID1BKD_HCT116_WT_HCT116.orig.txt.summary\" \u001b[0m                    ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "contrast: ARID1AKO_ARID1BKD_HCT116_ARID1AKO_HCT116\n",
      "test_group: ARID1AKO_ARID1BKD_HCT116 -> ['SRR5876164', 'SRR5876165']\n",
      "sample_negative_ctrl_grp: ARID1AKO_HCT116 -> ['SRR5876162', 'SRR5876163']\n",
      "\n",
      "\n",
      "       \u001b[44;37m ========== \u001b[0m\u001b[36m    _____ _    _ ____  _____  ______          _____  \n",
      "       \u001b[44;37m =====      \u001b[0m\u001b[36m   / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "       \u001b[44;37m   =====    \u001b[0m\u001b[36m  | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "       \u001b[44;37m     ====   \u001b[0m\u001b[36m   \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "       \u001b[44;37m       ==== \u001b[0m\u001b[36m   ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "       \u001b[44;37m ========== \u001b[0m\u001b[36m  |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\u001b[0m\n",
      "\t  v2.0.1\n",
      "\n",
      "//==========================\u001b[36m featureCounts setting \u001b[0m===========================\\\\\n",
      "||  \u001b[0m                                                                          ||\n",
      "||             Input files : \u001b[36m4 BAM files  \u001b[0m \u001b[0m                                   ||\n",
      "||                           \u001b[32mo\u001b[36m SRR5876164.small.chrsorted.BAM\u001b[0m \u001b[0m                ||\n",
      "||                           \u001b[32mo\u001b[36m SRR5876165.small.chrsorted.BAM\u001b[0m \u001b[0m                ||\n",
      "||                           \u001b[32mo\u001b[36m SRR5876162.small.chrsorted.BAM\u001b[0m \u001b[0m                ||\n",
      "||                           \u001b[32mo\u001b[36m SRR5876163.small.chrsorted.BAM\u001b[0m \u001b[0m                ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "||             Output file : \u001b[36mARID1AKO_ARID1BKD_HCT116_ARID1AKO_HCT116.ori\u001b[0m ...\u001b[0m \u001b[0m||\n",
      "||                 Summary : \u001b[36mARID1AKO_ARID1BKD_HCT116_ARID1AKO_HCT116.ori\u001b[0m ...\u001b[0m \u001b[0m||\n",
      "||              Annotation : \u001b[36mARID1AKO_ARID1BKD_HCT116_ARID1AKO_HCT116.mer\u001b[0m ...\u001b[0m \u001b[0m||\n",
      "||      Dir for temp files : \u001b[36m../merged_IDR_narrowPeak\u001b[0m \u001b[0m                        ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "||                 Threads : \u001b[36m1\u001b[0m \u001b[0m                                               ||\n",
      "||                   Level : \u001b[36mmeta-feature level\u001b[0m \u001b[0m                              ||\n",
      "||              Paired-end : \u001b[36mno\u001b[0m \u001b[0m                                              ||\n",
      "||      Multimapping reads : \u001b[36mnot counted\u001b[0m \u001b[0m                                     ||\n",
      "|| Multi-overlapping reads : \u001b[36mnot counted\u001b[0m \u001b[0m                                     ||\n",
      "||   Min overlapping bases : \u001b[36m1\u001b[0m \u001b[0m                                               ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "//=================================\u001b[36m Running \u001b[0m==================================\\\\\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Load annotation file ARID1AKO_ARID1BKD_HCT116_ARID1AKO_HCT116.merged.s\u001b[0m ... \u001b[0m||\n",
      "||    Features : \u001b[36m70760\u001b[0m \u001b[0m                                                       ||\n",
      "||    Meta-features : \u001b[36m70760\u001b[0m \u001b[0m                                                  ||\n",
      "||    Chromosomes/contigs : \u001b[36m24\u001b[0m \u001b[0m                                               ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file SRR5876164.small.chrsorted.BAM... \u001b[0m                        ||\n",
      "||    WARNING: \u001b[36mPaired-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m12931920\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m3191021 (24.7%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m0.84 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file SRR5876165.small.chrsorted.BAM... \u001b[0m                        ||\n",
      "||    WARNING: \u001b[36mPaired-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m5822936\u001b[0m \u001b[0m                                             ||\n",
      "||    Successfully assigned alignments : \u001b[36m1746883 (30.0%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m0.33 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file SRR5876162.small.chrsorted.BAM... \u001b[0m                        ||\n",
      "||    WARNING: \u001b[36mPaired-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m13048224\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m3359876 (25.7%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m0.85 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file SRR5876163.small.chrsorted.BAM... \u001b[0m                        ||\n",
      "||    WARNING: \u001b[36mPaired-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m7156354\u001b[0m \u001b[0m                                             ||\n",
      "||    Successfully assigned alignments : \u001b[36m2630099 (36.8%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m0.40 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Write the final count table. \u001b[0m                                              ||\n",
      "|| Write the read assignment summary. \u001b[0m                                        ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Summary of counting results can be found in file \"../merged_IDR_narrowPea \u001b[0m ||\n",
      "|| k/ARID1AKO_ARID1BKD_HCT116_ARID1AKO_HCT116.orig.txt.summary\" \u001b[0m              ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "contrast: ARID1BKD_TOV21G_WT_TOV21G\n",
      "test_group: ARID1BKD_TOV21G -> ['SRR5876663', 'SRR5876664']\n",
      "sample_negative_ctrl_grp: WT_TOV21G -> ['SRR5876661', 'SRR5876662']\n",
      "\n",
      "\n",
      "       \u001b[44;37m ========== \u001b[0m\u001b[36m    _____ _    _ ____  _____  ______          _____  \n",
      "       \u001b[44;37m =====      \u001b[0m\u001b[36m   / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "       \u001b[44;37m   =====    \u001b[0m\u001b[36m  | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "       \u001b[44;37m     ====   \u001b[0m\u001b[36m   \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "       \u001b[44;37m       ==== \u001b[0m\u001b[36m   ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "       \u001b[44;37m ========== \u001b[0m\u001b[36m  |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\u001b[0m\n",
      "\t  v2.0.1\n",
      "\n",
      "//==========================\u001b[36m featureCounts setting \u001b[0m===========================\\\\\n",
      "||  \u001b[0m                                                                          ||\n",
      "||             Input files : \u001b[36m4 BAM files  \u001b[0m \u001b[0m                                   ||\n",
      "||                           \u001b[32mo\u001b[36m SRR5876663.small.chrsorted.BAM\u001b[0m \u001b[0m                ||\n",
      "||                           \u001b[32mo\u001b[36m SRR5876664.small.chrsorted.BAM\u001b[0m \u001b[0m                ||\n",
      "||                           \u001b[32mo\u001b[36m SRR5876661.small.chrsorted.BAM\u001b[0m \u001b[0m                ||\n",
      "||                           \u001b[32mo\u001b[36m SRR5876662.small.chrsorted.BAM\u001b[0m \u001b[0m                ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "||             Output file : \u001b[36mARID1BKD_TOV21G_WT_TOV21G.orig.txt\u001b[0m \u001b[0m              ||\n",
      "||                 Summary : \u001b[36mARID1BKD_TOV21G_WT_TOV21G.orig.txt.summary\u001b[0m \u001b[0m      ||\n",
      "||              Annotation : \u001b[36mARID1BKD_TOV21G_WT_TOV21G.merged.saf (SAF)\u001b[0m \u001b[0m      ||\n",
      "||      Dir for temp files : \u001b[36m../merged_IDR_narrowPeak\u001b[0m \u001b[0m                        ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "||                 Threads : \u001b[36m1\u001b[0m \u001b[0m                                               ||\n",
      "||                   Level : \u001b[36mmeta-feature level\u001b[0m \u001b[0m                              ||\n",
      "||              Paired-end : \u001b[36mno\u001b[0m \u001b[0m                                              ||\n",
      "||      Multimapping reads : \u001b[36mnot counted\u001b[0m \u001b[0m                                     ||\n",
      "|| Multi-overlapping reads : \u001b[36mnot counted\u001b[0m \u001b[0m                                     ||\n",
      "||   Min overlapping bases : \u001b[36m1\u001b[0m \u001b[0m                                               ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "//=================================\u001b[36m Running \u001b[0m==================================\\\\\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Load annotation file ARID1BKD_TOV21G_WT_TOV21G.merged.saf \u001b[0m... \u001b[0m             ||\n",
      "||    Features : \u001b[36m64683\u001b[0m \u001b[0m                                                       ||\n",
      "||    Meta-features : \u001b[36m64683\u001b[0m \u001b[0m                                                  ||\n",
      "||    Chromosomes/contigs : \u001b[36m24\u001b[0m \u001b[0m                                               ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file SRR5876663.small.chrsorted.BAM... \u001b[0m                        ||\n",
      "||    WARNING: \u001b[36mPaired-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m14775364\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m2533101 (17.1%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m0.93 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file SRR5876664.small.chrsorted.BAM... \u001b[0m                        ||\n",
      "||    WARNING: \u001b[36mPaired-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m27175488\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m2336225 (8.6%)\u001b[0m \u001b[0m                      ||\n",
      "||    Running time : \u001b[36m1.75 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file SRR5876661.small.chrsorted.BAM... \u001b[0m                        ||\n",
      "||    WARNING: \u001b[36mPaired-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m10292336\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m1853334 (18.0%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m0.64 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file SRR5876662.small.chrsorted.BAM... \u001b[0m                        ||\n",
      "||    WARNING: \u001b[36mPaired-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m28519554\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m2020936 (7.1%)\u001b[0m \u001b[0m                      ||\n",
      "||    Running time : \u001b[36m1.83 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Write the final count table. \u001b[0m                                              ||\n",
      "|| Write the read assignment summary. \u001b[0m                                        ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Summary of counting results can be found in file \"../merged_IDR_narrowPea \u001b[0m ||\n",
      "|| k/ARID1BKD_TOV21G_WT_TOV21G.orig.txt.summary\" \u001b[0m                             ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "\\\\============================================================================//\n",
      "\n"
     ]
    }
   ],
   "source": [
    "def compute_featureCounts(contrast_dict):    \n",
    "\n",
    "    contrast_list = list(set(contrast_dict.keys()))\n",
    "    print(\"\\ncontrast_list: \\n{}\\n\".format(contrast_list))\n",
    "\n",
    "    for contrast in list(contrast_dict.keys()):\n",
    "            print(\"contrast: {}\".format(contrast))\n",
    "            test_group = list(contrast_dict[contrast]['test_group'].keys())[0]\n",
    "            sample_test_group = list(contrast_dict[contrast]['test_group'].values())[0]\n",
    "            print('test_group: {} -> {}'.format(test_group, sample_test_group))\n",
    "\n",
    "            negative_ctrl_grp = list(contrast_dict[contrast]['negative_ctrl_grp'].keys())[0]\n",
    "            sample_negative_ctrl_grp = list(contrast_dict[contrast]['negative_ctrl_grp'].values())[0]\n",
    "            print(\"sample_negative_ctrl_grp: {} -> {}\\n\".format(negative_ctrl_grp, sample_negative_ctrl_grp))\n",
    "\n",
    "\n",
    "            # 7) Call featureCounts on saf file for each contrast      \n",
    "            bam_list = [BAM_DIR + group + '.small.chrsorted.BAM' for group in sample_test_group + sample_negative_ctrl_grp]\n",
    "            bam_list_bash = '\\t'.join(bam_list)\n",
    "            cmd_featureCounts = \"\"\"featureCounts -a {merged_IDR_narrowPeak_dir}{contrast}.merged.saf -F SAF  -o {merged_IDR_narrowPeak_dir}{contrast}.orig.txt {bam_list_bash}\"\"\".format(contrast=contrast, merged_IDR_narrowPeak_dir=merged_IDR_narrowPeak_dir, bam_list_bash=bam_list_bash)\n",
    "            # print(cmd_featureCounts)\n",
    "            ! {cmd_featureCounts}\n",
    "\n",
    "            # 8) Remove header \n",
    "            !sed -i '1,2d' {merged_IDR_narrowPeak_dir}{contrast}.orig.txt\n",
    "\n",
    "            # 9) reformat header for DPA logFC computation\n",
    "            sample_list_bash = '\\t'.join(sample_test_group + sample_negative_ctrl_grp)\n",
    "            sample_number = [1,2,3,4] + list(np.arange(7, 7+ len(sample_test_group + sample_negative_ctrl_grp), 1))\n",
    "            sample_number_str = [str(i) for i in sample_number]\n",
    "            sample_number_bash = '$' + ', $'.join(sample_number_str)\n",
    "            cmd_header = \"\"\"awk 'BEGIN{{FS=OFS=\"\\t\"; print \"peak_id\\tchr\\tstart\\tend\\t{sample_list_bash}\"}} {{print {sample_number_bash}}}' {merged_IDR_narrowPeak_dir}{contrast}.orig.txt > {featureCounts_dir}{contrast}.txt\"\"\".format(sample_list_bash=sample_list_bash, sample_number_bash=sample_number_bash, merged_IDR_narrowPeak_dir=merged_IDR_narrowPeak_dir,featureCounts_dir=featureCounts_dir, contrast=contrast)\n",
    "            # print(cmd_header)\n",
    "            ! {cmd_header}\n",
    "            \n",
    "compute_featureCounts(contrast_dict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0c7d1313-696e-4122-9e4e-776fb6cfeaee",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8ccb64ad-05f1-43ef-9244-5e46a756fcf3",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "87cff33e-ab3a-4ea5-bc06-1be4fa668b0b",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
